python/taichi/lang/ops.py

```diff
@@ -181,68 +206,117 @@ def _ternary_operation(taichi_op, python_op, a, b, c):
+        x (Union[:mod:`~taichi.types.primitive_types`, :class:`~taichi.Matrix`]): \
+            Input scalar or matrix.
 
     Returns:
-        The negative value of `a`.
+        Matrix or scalar `y`, so that `y = -x`. This is a scalar if `x` is a scalar.
+
+    Example::
+        >>> x = ti.Matrix([1, -1])
+        >>> y = ti.neg(a)
+        >>> y
+        [-1, 1]
     """
-    return _unary_operation(_ti_core.expr_neg, _bt_ops_mod.neg, a)
+    return _unary_operation(_ti_core.expr_neg, _bt_ops_mod.neg, x)
 
 
 @unary
-def sin(a):
-    """The sine function.
+def sin(x):
+    """Trigonometric sine, element-wise.
 
     Args:
-        a (Union[:class:`~taichi.lang.expr.Expr`, :class:`~taichi.lang.matrix.Matrix`]): A number or a matrix.
+        x (Union[:mod:`~taichi.types.primitive_types`, :class:`~taichi.Matrix`]): \
+            Angle, in radians.
 
     Returns:
-        Sine of `a`.
+        The sine of each element of `x`.
+
+    Example::
+
+        >>> from math import pi
+        >>> x = ti.Matrix([-pi/2., 0, pi/2.])
+        >>> ti.sin(x)
+        [-1., 0., 1.]
     """
-    return _unary_operation(_ti_core.expr_sin, math.sin, a)
+    return _unary_operation(_ti_core.expr_sin, math.sin, x)
 
 
 @unary
-def cos(a):
-    """The cosine function.
+def cos(x):
+    """Trigonometric cosine, element-wise.
 
     Args:
-        a (Union[:class:`~taichi.lang.expr.Expr`, :class:`~taichi.lang.matrix.Matrix`]): A number or a matrix.
+        x (Union[:mod:`~taichi.type.primitive_types`, :class:`~taichi.Matrix`]): \
+            Angle, in radians.
 
     Returns:
-        Cosine of `a`.
+        The cosine of each element of `x`.
+
+    Example::
+
+        >>> from math import pi
+        >>> x = ti.Matrix([-pi, 0, pi/2.])
+        >>> ti.cos(x)
+        [-1., 1., 0.]
     """
-    return _unary_operation(_ti_core.expr_cos, math.cos, a)
+    return _unary_operation(_ti_core.expr_cos, math.cos, x)
 
 
 @unary
-def asin(a):
-    """The inverses function of sine.
+def asin(x):
+    """Trigonometric inverse sine, element-wise.
+
+    The inverse of `sin` so that, if `y = sin(x)`, then `x = asin(y)`.
+
+    For input `x` not in the domain `[-1, 1]`, this function returns `nan` if \
+        it's called in taichi scope, or raises exception if it's called in python scope.
 
     Args:
-        a (Union[:class:`~taichi.lang.expr.Expr`, :class:`~taichi.lang.matrix.Matrix`]): A number or a matrix with elements in [-1,1].
+        x (Union[:mod:`~taichi.types.primitive_types`, :class:`~taichi.Matrix`]): \
+            A scalar or a matrix with elements in [-1, 1].
 
     Returns:
-        The inverses function of sine of `a`.
+        The inverse sine of each element in `x`, in radians and in the closed \
+            interval `[-pi/2, pi/2]`.
+
+    Example::
+
+        >>> from math import pi
+        >>> ti.asin(ti.Matrix([-1.0, 0.0, 1.0])) * 180 / pi
+        >>> [-90., 0., 90.]
```

Comment:
Aha I finally got what feels a bit weird here: you only need `>>>` for commands but not outputs so that users can tell the difference ;) 
Examples: https://github.com/google/jax/blob/4e219220558a2279de2abc9dc4140dedeb61703f/jax/_src/errors.py#L96
https://github.com/pytorch/pytorch/blob/master/torch/functional.py#L139